analysis/shamanenhancement/src/modules/shadowlands/legendaries/WitchDoctorsWolfBones.tsx

```diff
@@ -0,0 +1,90 @@
+import SPELLS from 'common/SPELLS';
+import CooldownIcon from 'interface/icons/Cooldown';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  ApplyBuffEvent,
+  ApplyBuffStackEvent,
+  EventType,
+  RefreshBuffEvent,
+} from 'parser/core/Events';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+class WitchDoctorsWolfBones extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  protected spellUsable!: SpellUsable;
+  protected lastTimestamp = 0;
+  protected hasApplyBuffInThisTimestamp = false;
+  protected totalCdrGained = 0;
+  protected totalCdrWasted = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasLegendary(SPELLS.WITCH_DOCTORS_WOLF_BONES);
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.MAELSTROM_WEAPON_BUFF),
+      this.reduceFeralSpirityCooldown,
+    );
+
+    this.addEventListener(
+      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.MAELSTROM_WEAPON_BUFF),
+      this.reduceFeralSpirityCooldown,
+    );
+
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.MAELSTROM_WEAPON_BUFF),
+      this.reduceFeralSpirityCooldown,
+    );
+  }
+
+  reduceFeralSpirityCooldown(event: ApplyBuffStackEvent | ApplyBuffEvent | RefreshBuffEvent) {
+    if (event.timestamp !== this.lastTimestamp) {
+      this.hasApplyBuffInThisTimestamp = false;
+    }
+
+    if (event.type === EventType.RefreshBuff && this.hasApplyBuffInThisTimestamp) {
+      return;
+    }
+
+    if (event.type === EventType.ApplyBuff || event.type === EventType.ApplyBuffStack) {
+      this.hasApplyBuffInThisTimestamp = true;
+    }
+
+    if (this.spellUsable.isOnCooldown(SPELLS.FERAL_SPIRIT.id)) {
+      this.spellUsable.reduceCooldown(SPELLS.FERAL_SPIRIT.id, 2000, event.timestamp);
+      this.totalCdrGained += 2000;
+    } else {
+      this.totalCdrWasted += 2000;
+    }
+
+    this.lastTimestamp = event.timestamp;
+  }
+
+  statistic() {
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.OPTIONAL(10)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.ITEMS}
+      >
+        <BoringSpellValueText spellId={SPELLS.WITCH_DOCTORS_WOLF_BONES.id}>
+          <>
+            <CooldownIcon /> {this.totalCdrGained / 1000}s <small> of Feral Spirit CDR</small>
```

Comment:
Should use `formatDuration` here

Reply:
Both places are using formatDuration now